Center on clicked symbol, addresses #1516

danswick opened the pull request:
cc: @lyzidiamond 

Would love some feedback on this example. Does it adequately replicate [this](https://www.mapbox.com/mapbox.js/example/v1.0.0/centering-markers/) popular Mapbox.js example?

![marker-center](https://cloud.githubusercontent.com/assets/2365503/10036294/74d16642-615a-11e5-85a1-668d9e5719f6.gif)

jfirebaugh commented on docs/_posts/examples/3400-01-04-center-on-marker.html:
```diff
@@ -0,0 +1,95 @@
+});
+
+map.on('click', function(e){
+  map.featuresAt(e.point, { radius: 20, includeGeometry: true }, function(err, features){
```
```
    map.featuresAt(e.point, {layer: 'myMarkers', radius: 20, includeGeometry: true}, function (err, features) {
        if (err) throw err;
        if (features.length) {
            map.flyTo({center: features[0].geometry.coordinates});
        }
    });
```

* Uses the `layer` option for `featuresAt` to avoid getting results for any other layers
* Uses the coordinates from the marker itself, like the Mapbox.js example

Reference #1002 for a future API enhancement that would simplify this example to:

```
map.on('click', 'myMarkers', function (e) {
    map.flyTo({center: e.features[0].geometry.coordinates});
});
```

danswick commented:
Thanks for the note, @jfirebaugh! Added your suggestions and cursor change on hover to indicate markers are clickable (thanks, @edenh!). 

Also just want to take everyone's temperature on terminology here. There isn't really a "marker" in GL, so I renamed this example "Center on symbol" to stick with GL terms. Does that make sense or is it even more confusing? I considered "Center on point," but that may be even worse. cc: @lyzidiamond.

bhousel commented:
Phrases like "marker", "layer", "control" are terms of art in Mapbox.js / Leaflet development, and I think it's best to carry them forward when speaking about Mapbox GL as well.

jfirebaugh commented:
Committed to `mb-pages` in 28eca5ab15cdc7193e78331056e374f6dafbeaac. Thanks!
